pkg/controller/nebulabackup/nebula_backup_manager.go

```diff
@@ -118,65 +140,346 @@ func (bm *backupManager) Sync(backup *v1alpha1.NebulaBackup) error {
 										},
 									},
+									{
+										Name: EnvS3SecretKeyName,
+										ValueFrom: &corev1.EnvVarSource{
+											SecretKeyRef: &corev1.SecretKeySelector{
+												LocalObjectReference: corev1.LocalObjectReference{
+													Name: backup.Spec.BR.S3.SecretName,
+												},
+												Key: br.S3SecretKey,
+											},
+										},
+									},
+								},
+								Command: []string{"/bin/bash", "-c"},
+								Args:    []string{backupCmd},
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										Name:      "backup-cleanup-volume-runnable",
+										MountPath: "/usr/local/bin/runnable",
+									},
+								},
+							},
+						},
+						InitContainers: []corev1.Container{
+							{
+								Name:            "backup-init",
+								Image:           backup.Spec.ToolImage,
+								ImagePullPolicy: backup.Spec.ImagePullPolicy,
+								Command:         []string{"/bin/bash", "-c"},
+								Args:            []string{"cp /usr/local/bin/backup-cleanup.sh /usr/local/bin/runnable/backup-cleanup-run.sh; chmod +x /usr/local/bin/runnable/backup-cleanup-run.sh"},
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										Name:      "backup-cleanup-volume",
+										MountPath: "/usr/local/bin/backup-cleanup.sh",
+										SubPath:   "backup-cleanup.sh",
+									},
+									{
+										Name:      "backup-cleanup-volume-runnable",
+										MountPath: "/usr/local/bin/runnable",
+									},
+								},
+							},
+						},
+						Volumes: []corev1.Volume{
+							{
+								Name: "backup-cleanup-volume",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{Name: getScheduledBackupScriptsObjectKey(backup).Name},
+									},
+								},
+							},
+							{
+								Name: "backup-cleanup-volume-runnable",
+								VolumeSource: corev1.VolumeSource{
+									EmptyDir: &corev1.EmptyDirVolumeSource{},
 								},
-								{
-									Name: EnvS3SecretKeyName,
-									ValueFrom: &corev1.EnvVarSource{
-										SecretKeyRef: &corev1.SecretKeySelector{
-											LocalObjectReference: corev1.LocalObjectReference{
-												Name: backup.Spec.BR.S3.SecretName,
+							},
+						},
+						RestartPolicy: corev1.RestartPolicyNever,
+						NodeSelector:  backup.Spec.NodeSelector,
+					},
+				},
+				TTLSecondsAfterFinished: aws.Int32(600),
+			},
+		}
+	} else {
+		bpCmdHead := "exec /usr/local/bin/br-ent backup full"
+		if backup.Spec.BR.BackupName != "" {
+			bpCmdHead = fmt.Sprintf("exec /usr/local/bin/br-ent backup incr --base %s", backup.Spec.BR.BackupName)
+		}
+
+		backupCmd := fmt.Sprintf("%s --meta %s --storage s3://%s --s3.access_key $%s --s3.secret_key $%s --s3.region %s --s3.endpoint %s",
+			bpCmdHead, metaAddr, backup.Spec.BR.S3.Bucket, EnvS3AccessKeyName, EnvS3SecretKeyName, backup.Spec.BR.S3.Region, backup.Spec.BR.S3.Endpoint)
+
+		return &batchv1.Job{
```

Comment:
OwnerReference is lost

Reply:
@MegaByte875 Got it, will add a owner reference when creating this.

Reply:
Fixed